### PR TITLE
feat: nomad development environment

### PR DESCRIPTION
**Pull request by lukasmetzner**
This PR adds the initial version of the nomad-dev-env setup.

Following tasks need to be completed in order for the PR to be ready:

- [x] lint in ci
- [x] test setup in ci
- [x] setup releaser pleaser
- [x] setup renovate

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,50 @@
+---
```
Comment (jooola) [inline .pre-commit-config.yaml:1]: You have a bunch of shell scripts, might be worth to add shell check as well :)


https://github.com/hetznercloud/hcloud-cloud-controller-manager/blob/dedc259439f46773743668c01a7b6dfcc1711995/.pre-commit-config.yaml#L39-L52

Comment (lukasmetzner) [reply .pre-commit-config.yaml:1]: Sounds good. Added them

File: variables.tf [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  default = "1.9.3"
+}
+
+variable "worker_count" {
+  default = 1
+}
+
+variable "server_type" {
+  default = "cx22"
+}
```
Comment (apricote) [inline variables.tf:19]: For parity with `kubernetes-dev-env`, could you rename this variable to `hcloud_server_type` and add the following variables and their behaviour?

- `name`
- `hcloud_location`
- `hcloud_labels`

File: main-setup.tf [hunk 3]
```diff
@@ -0,0 +1,282 @@
...
+  provisioner "local-exec" {
+    command = "mkdir -p  ${abspath("${path.root}/files/")}"
+  }
+}
+
+resource "terraform_data" "certificates" {
+  provisioner "local-exec" {
+    command = "bash ${abspath("${path.module}/scripts/generate-tls-certs.sh")} ${abspath("${path.root}/files/")} ${hcloud_server.control.ipv4_address}"
+  }
+}
```
Comment (apricote) [inline main-setup.tf:11]: Is the `mkdir` really necessary? Right now there is no clear dependency from `terraform_data.certificates` on `terraform_data.make_files_dir` to make sure that the directory was actually created before generating the certs.

**Review by apricote — COMMENTED** [review]
:rocket:

File: variables.tf (hunk 2 above)
Comment (lukasmetzner) [reply variables.tf]: Done

**Review by apricote — APPROVED** [review]

File: main-setup.tf (hunk 3 above)
Comment (lukasmetzner) [reply main-setup.tf]: I have tested it quickly and actually it is not necessary, as the folder is created through the `local_sensitive_file` resource when generating the ssh key pair.

**Pull request merged**